fedimint-server/src/consensus/engine.rs

```diff
@@ -350,7 +350,10 @@ impl ConsensusEngine {
                         .items
                         .split_at(pending_accepted_items.len());
 
-                    assert!(processed.iter().eq(pending_accepted_items.iter()));
+                    assert!(
+                        processed.iter().eq(pending_accepted_items.iter()),
+                        "Consensus Failure: pending accepted items disagree with federation consensus"
+                    );
```

Comment:
If this was an `assert_eq` (over a `Vec`, idk about iterators) we would see the difference.